### ISSUE TITLE
[Feature] Custom Components

### ISSUE THREAD

**Issue by dak** [hankinsoft/MudClient#13]
Concept: As a user, I'd like to be able to create and style visual components that I can interact with and can display state information.

Possible Implementation: Allow the creation of one or more resizable WebView components that can be repositioned around the client itself that can be edited with HTML/CSS/JavaScript and has access to the special variables and commands of the client. Preferably provide some sort of two-way binding with client variables as well, or barring that, methods to update client variables from the component and to force the component to redraw from a trigger or alias.

**Comment by hankinsoft** [comment]
How are you envisioning you would create the html, css, etc? I'm thinking of something like the triggers, aliases, notes window, where the left side has the 'component' and the right side contains the default html/css. Then you can update via scripts in triggers/aliases?

**Comment by dak** [comment]
Yeah, I don't think it would be unreasonable to expect the JavaScript/HTML/CSS all be inlined with a similar UI to the triggers and aliases. The JavaScript should have access to the same scope/variables as the triggers and aliases, and then just render it in a WebView. I suppose the WebView should preferably also have a way to set/save positioning and sizing as well.

For the positioning, it would be amazing if you could also position and size WebViews either absolutely on the screen or relative to the client view (with relative positioning being "sticky" if the client view is dragged/resized).

**Comment by dak** [comment]
https://quow.co.uk/mushclient.php is a good example of how MUSH Client does this, and how it's used.

**Comment by hankinsoft** [comment]
Are there any specific UI elements you're looking to see added? I've been investigating this a bit, but before I put up any documentation (for code that may change) I'd like to get an example and make sure my implementation can handle it.
